File: quantificadores/naoGuloso.js
const texto = '<div>Conteudo 01</div><div>Conteudo 02</div>'

// quantificadores são greedy

console.log(texto.match(/<div>.+<\/div>/g))
console.log(texto.match(/<div>.*<\/div>/g))
console.log(texto.match(/<div>.{0,100}<\/div>/g))

// quantificadoes não greedy
console.log(texto.match(/<div>.+?<\/div>/g))  // ? torna a regez lazy (não greedy)
console.log(texto.match(/<div>.*?<\/div>/g))
console.log(texto.match(/<div>.{0,100}?<\/div>/g))

File: quantificadores/naoGuloso.py
import re

texto = '<div>Conteudo 01</div><div>Conteudo 02</div>'

print(re.findall('<div>.+<\/div>', texto))
print(re.findall('<div>.*<\/div>', texto))
print(re.findall('<div>.{0,100}<\/div>', texto))

print(re.findall('<div>.+?<\/div>', texto))
print(re.findall('<div>.*?<\/div>', texto))
print(re.findall('<div>.{0,100}?<\/div>', texto))